packages/manager/apps/pci-savings-plan/src/utils/kpi/utils.ts

```diff
@@ -1,17 +1,87 @@
 import { SavingsPlanFlavorConsumption } from '@/types/savingsPlanConsumption.type';
 
+export enum Period {
+  utilization = 'utilization',
+  coverage = 'coverage',
+}
+
 export const getPercentValue = (
   consumption: SavingsPlanFlavorConsumption,
-  key: 'utilization' | 'coverage',
-): string => {
+  key: Period.utilization | Period.coverage,
+): number => {
   const values = consumption.periods
     .map((period) => parseFloat(period[key].replace('%', '')))
     .filter((value) => !Number.isNaN(value));
 
   if (!values.length) {
-    return '';
+    return 0;
   }
 
   const average = values.reduce((sum, value) => sum + value, 0) / values.length;
-  return `${Math.floor(average)}%`;
+  return Math.floor(average);
+};
+
+export const isCurrentPeriod = (period: string): boolean => {
+  const now = new Date();
+  const periodDate = new Date(period);
+
+  if (Number.isNaN(periodDate.getTime())) {
+    return false;
+  }
+
+  return (
+    periodDate.getFullYear() === now.getFullYear() &&
+    periodDate.getMonth() === now.getMonth()
+  );
+};
+
+export const calculateComputedPercents = (
+  consumption: SavingsPlanFlavorConsumption,
+  formatPrice: (
+    price: number,
+    options: { decimals: number; unit: number },
+  ) => string,
+) => {
+  const { fees } = consumption;
+  const { saved_amount: rawSavedAmount = 0, total_price: rawTotalPrice = 0 } =
+    fees ?? {};
+
+  if (!consumption.periods?.length) {
+    return {
+      computedSavedAmount: null,
+      computedWithoutSavedAmount: null,
+    };
+  }
+
+  const savedAmount = Number(rawSavedAmount);
+  const totalPrice = Number(rawTotalPrice);
+
+  const totalAmountWithoutSavingsPlans =
+    Math.min(savedAmount, totalPrice) <= 0 ? 0 : savedAmount + totalPrice;
+
+  const computedSavedAmount =
+    savedAmount > 0 ? formatPrice(savedAmount, { decimals: 2, unit: 1 }) : null;
+
+  const computedWithoutSavedAmount =
+    totalAmountWithoutSavingsPlans > 0
+      ? formatPrice(totalAmountWithoutSavingsPlans, { decimals: 2, unit: 1 })
+      : null;
+
+  return { computedSavedAmount, computedWithoutSavedAmount };
+};
+
+export const formatKpiValue = (
+  value: number | string,
+  titleTranslationKey: string,
+): string => {
+  if (!value) {
+    return 'dashboard_kpis_not_available';
+  }
+  const percentageKeys = [
+    'dashboard_kpis_usage_percent_name',
+    'dashboard_kpis_coverage_percent_name',
+  ];
+  return percentageKeys.includes(titleTranslationKey)
```

Comment:
I don't understand what you are checking here. Also the titleTranslationKey isn't used for anything else